### PR TITLE
T2080c

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3b9ec9c8bb8e164/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/3b9ec9c8bb8e164/output.txt

Total script time: 1.12 mins

Server is running on http://pr908.butyr.org/templates/basic/
Run the Test Suite: http://pr908.butyr.org/test/

File: templates/basic/template.js [hunk 1]
```diff
@@ -8,6 +8,15 @@
...
       ready: function( butter ) {
         EditorHelper( butter );
+
+        function addTabzillaCode() {
+          var script = document.createElement( "script" );
+          script.src = "//www.mozilla.org/tabzilla/media/js/tabzilla.js";
+          document.body.appendChild( script );
+        }
+
+        addTabzillaCode();
```
Comment (secretrobotron) [inline templates/basic/template.js]: Does this need its own function? This code can probably just run inline.

File: css/header.less [hunk 2]
```diff
@@ -19,52 +19,201 @@
...
+  background: url( "../resources/tabzilla.png" );
+  height: 44px;
+  width: 150px;
+  position: absolute;
+  top: 0;
+  right: 30px;
+  text-indent: -2000px;
+  overflow: hidden;
+  z-index: 999;
+}
```
Comment (secretrobotron) [inline css/header.less]: I think much of this block isn't required anymore, since CSS is provided by the tabzilla stylesheet. Just keep the pieces that place the tab where we want.

File: src/layouts/header.html [hunk 3]
```diff
@@ -1,12 +1,26 @@
 <div class="butter-header" data-butter-exclude>
+  <div class="tabzilla-area"></div>
```
Comment (secretrobotron) [inline src/layouts/header.html]: Probably don't need this anymore, right?

File: css/header.less (hunk 2 above)
Comment (k88hudson) [inline css/header.less]: Some of it is necessary since it's different from the default CSS but I'll remove everything that's duplication

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/0fefa062e32d594/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/0fefa062e32d594/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/f1c3547db292566/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/f1c3547db292566/output.txt

Total script time: 1.68 mins

Server is running on http://pr908.butyr.org/templates/basic/
Run the Test Suite: http://pr908.butyr.org/test/

File: src/ui/header.js [hunk 4]
```diff
@@ -16,19 +16,32 @@ define([ "dialog/dialog", "util/lang", "ui/user-data", "ui/widget/tooltip" ],
...
+    _rootElement.querySelector( ".butter-logo" ).addEventListener( "dblclick", function( e ) {
+      butter.editor.openEditor( "ui-kit" );
+    }, false );
+
     _this.element = _rootElement;
-    ToolTip.apply( _projectTitle );
+
+    _tabzilla.addEventListener( "click", function( e ) {
+      e.preventDefault();
+      document.body.classList.toggle( "tabzilla-open");
```
Comment (secretrobotron) [inline src/ui/header.js]: Spacing.

File: templates/basic/template.js [hunk 5]
```diff
@@ -7,7 +7,10 @@
     Butter({
       config: "config.json",
       ready: function( butter ) {
+        var script = document.createElement( "script" );
```
Comment (secretrobotron) [inline templates/basic/template.js]: I like what you attempted to do here, but it should probably look like this:

``` javascript
var script;
EditorHelper( butter );
script = document.createElement( "script" );
script.src = "...";
...
```

File: src/ui/header.js [hunk 6]
```diff
@@ -16,19 +16,32 @@ define([ "dialog/dialog", "util/lang", "ui/user-data", "ui/widget/tooltip" ],
...
+    // This is an easter egg to open a UI kit editor. Hurrah
+    _rootElement.querySelector( ".butter-logo" ).addEventListener( "dblclick", function( e ) {
+      butter.editor.openEditor( "ui-kit" );
+    }, false );
+
     _this.element = _rootElement;
-    ToolTip.apply( _projectTitle );
+
+    _tabzilla.addEventListener( "click", function( e ) {
+      e.preventDefault();
```
Comment (secretrobotron) [inline src/ui/header.js]: Why is this here? Does something conflict when it's not?

File: css/ui-states.less [hunk 7]
```diff
@@ -75,3 +75,32 @@
...
+.tabzilla-open {
+  &.butter-header-spacing {
+    padding-top: @TABZILLA_HEIGHT + @HEADER_SPACING;
+  }
+  .butter-header {
+    top: @TABZILLA_HEIGHT;
+    z-index: @MODAL_Z_INDEX;
+    box-shadow: 0 -1px 0 0 #CCC;
+  }
+  #tabzilla {
```
Comment (secretrobotron) [inline css/ui-states.less:107]: Not sure this declaration is needed at all. tabzilla has its own css, right? Do we conflict with it without this?

File: src/ui/header.js (hunk 6 above)
Comment (k88hudson) [inline src/ui/header.js]: Omitting the `href` atrribute is probably enough actually, as long as that works for all browsers

File: css/ui-states.less (hunk 7 above)
Comment (k88hudson) [inline css/ui-states.less:107]: This is just because there's a border on the top, and I don't want the tab to look cut-off when it's expanded

File: src/layouts/header.html [hunk 8]
```diff
@@ -1,12 +1,24 @@
...
+         <ul class="butter-dropdown-off">
+          <li><a class="butter-my-projects-btn" href="/dashboard">My Projects</a></li>
+          <li><a class="butter-logout-btn">Logout</a></li>
+        </ul>
       </div>
-      <a class="butter-btn btn-green butter-save-btn">Save</a>
+      </li>
+    </ul>
+
+    <div class="butter-header-buttons-right">
```
Comment (secretrobotron) [inline src/layouts/header.html]: I don't think this has a purpose here.

Comment (secretrobotron) [inline src/layouts/header.html]: (or its CSS above)

File: src/ui/header.js [hunk 9]
```diff
@@ -142,20 +154,26 @@ define([ "dialog/dialog", "util/lang", "ui/user-data", "ui/widget/tooltip" ],
...
-    _projectTitle.addEventListener( "click", projectNameClick, false );
 
+    _projectName.addEventListener( "click", projectNameClick, false );
+
+    function toggleDropDown() {
+      _nameDropDown.classList.toggle( "butter-dropdown-off" );
+    }
+  
     function doLogout() {
-      _userData.logout( logoutDisplay );
```
Comment (secretrobotron) [inline src/ui/header.js]: Why didn't you just keep this line instead of resetting the editor to `plugin-list`?

File: cornfield/views/dashboard.jade [hunk 10]
```diff
@@ -78,6 +78,9 @@ html
         font-size: .9em;
         overflow: scroll;
       }
+      .butter-logo {
+        background: url( "../resources/logo-words.png" ) no-repeat;
```
Comment (jbuck) [inline cornfield/views/dashboard.jade]: Move this logo to the public folder

File: css/buttons.less [hunk 11]
```diff
@@ -397,7 +397,8 @@
 .icon-exclamation-sign {
   background-position: 0 -120px;
 }
-.icon-gift {
+.icon-downtick {
+  margin-left: 0;
```
Comment (jbuck) [inline css/buttons.less:401]: ?

File: css/header.less [hunk 12]
```diff
@@ -40,31 +90,109 @@
 
 .butter-header {
   .butter-btn {
-    padding: 8px 10px;
```
Comment (jbuck) [inline css/header.less:43]: This looks weird, but could be right, I don't know.

File: css/buttons.less (hunk 11 above)
Comment (k88hudson) [inline css/buttons.less:401]: Typically we slide the icon over a bit but since this is a special one, we don't need it

File: css/header.less (hunk 12 above)
Comment (k88hudson) [inline css/header.less:43]: In the header, the buttons are no longer buttons really anymore (ie they dont have backgrounds or borders to deal with)

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5b335caa22ce44b/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5b335caa22ce44b/output.txt

Total script time: 0.82 mins

Server is running on http://pr908.butyr.org/templates/basic/
Run the Test Suite: http://pr908.butyr.org/test/
